compiler/rustc_middle/src/mir/syntax.rs

```diff
@@ -23,68 +23,78 @@ use crate::ty::{self, GenericArgsRef, List, Region, Ty, UserTypeAnnotationIndex}
 
 /// Represents the "flavors" of MIR.
 ///
-/// All flavors of MIR use the same data structure, but there are some important differences. These
-/// differences come in two forms: Dialects and phases.
+/// All flavors of MIR use the same data structure, but there are some important differences. Each
+/// flavor has (a) a dialect and, (b) a phase within that dialect. A single
+/// `MirPhase` value specifies both a dialect and a phase.
```

Comment:
This is a somewhat confusing way to put it IMO. I would start from the other side: "The MIR pipeline is structured into a few major dialects, and then multiple phases within each dialect. A MIR flavor is thus identified by a dialect-phase pair."

Reply:
ok